src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs

```diff
@@ -101,6 +106,11 @@ public sealed class TimestampType : AtomicType
         /// </summary>
         internal override object FromInternal(object obj)
         {
+            if (obj == null)
+            {
+                return null;
+            }
+
             // Known issue that if the original type is "long" and its value can be fit into the
             // "int", Pickler will serialize the value as int.
             if (obj is long val)
```

Comment:
```suggestion
            if (obj is long)
```

Reply:
`val` is needed.

Reply:
The way this if statement is structured is a bit weird, since you already have a casted `val` from the `(obj is long val)`, and in the if body you are casting obj again:
```
            if (obj is long val)
            {
                val = (long)obj;
            }
```

How about something like `long val = (obj is long v) ? v : (int)obj;` instead.

Reply:
Or maybe something like this:

```csharp
            long val = obj switch
            {
                int i => i,
                long l => l,
                _ => throw new Exception("..."),
            };
```

Reply:
I have updaded it accordingly . Thanks for your suggestions!